react/features/base/react/components/native/SlidingView.js

```diff
@@ -0,0 +1,241 @@
+// @flow
+
+import React, { PureComponent, type Node } from 'react';
+import {
+    Animated,
+    Dimensions,
+    TouchableWithoutFeedback,
+    View
+} from 'react-native';
+
+import { type StyleType } from '../../../styles';
+
+import styles from './slidingviewstyles';
+
+/**
+ * The type of the React {@code Component} props of {@link SideBar}.
+ */
+type Props = {
+
+    /**
+     * The children of {@code SideBar}.
+     */
+    children: Node,
+
+    /**
+     * Callback to notify the containing {@code Component} that the view is
+     * closing.
+     */
+    onHide: Function,
+
+    /**
+     * Position of the SlidingView: 'left', 'right', 'top', 'bottom'.
+     * later).
+     */
+    position: string,
+
+    /**
+     * Whether the menu (of the {@code SideBar}?) is displayed/rendered/shown.
+     */
+    show: boolean,
+
+    /**
+     * Style of the animated view.
+     */
+    style: StyleType
+};
+
+/**
+ * The type of the React {@code Component} state of {@link SideBar}.
+ */
+type State = {
+
+    /**
+     * Whether the sliding overlay should be displayed/rendered/shown.
+     */
+    showOverlay: boolean,
+
+    /**
+     * The native animation object.
+     */
+    sliderAnimation: Animated.Value,
+
+    /**
+     * Offset to move the view out of the screen.
+     */
+    positionOffset: number
+};
+
+/**
+ * A generic animated slider view to be used for animated menus.
+ */
+export default class SlidingView extends PureComponent<Props, State> {
+    /**
+     * Implements React's {@link Component#getDerivedStateFromProps()}.
+     *
+     * @inheritdoc
+     */
+    static getDerivedStateFromProps(props: Props, prevState: State) {
+        return {
+            showOverlay: props.show || prevState.showOverlay
+        };
+    }
+
+    /**
+     * Initializes a new {@code SideBar} instance.
+     *
+     * @inheritdoc
+     */
+    constructor(props: Props) {
+        super(props);
+
+        const { height, width } = Dimensions.get('window');
```

Comment:
Please avoid using `Dimensions` since the app view can change sized in multi-app mode (both on iOS and Android) so using `onLayout` is preferred. You can wrap your view in a `DimensionsDetector` (in features/base/responsive-ui) instead.

Reply:
I tried, but onLayout is not really suitable when a view is animated, as it gets updated too many times. Now I consider this a "safe" value, as we don't need the exact size of the app, we just need a value that is surely out of the screen, but not extreme, like 10000. Now with this explained, do you think it can cause issues?